app/WebParsing/UtsgJsonParser.hs

```diff
@@ -0,0 +1,251 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{-
+module WebParsing.UtsgJsonParser
+     (getJSON) where
+-}
+
+import Data.Aeson
+import Data.List
+import Database.Tables
+import WebParsing.PrerequisiteParsing
+import Data.Maybe
+import Control.Monad
+import Database.Persist.Sqlite (runSqlite, insertMany_, insert_)
+import qualified Data.HashMap.Lazy as M
+import qualified Data.Text as T
+import Config (databasePath)
+import Network.HTTP.Conduit (simpleHttp)
+
+-- | Converts 24-hour time into a double
+-- | Assumes times are rounded to the nearest hour
+getHourVal :: String -> Double
+getHourVal time = (read $ take 2 time :: Double) + (/) (read $ drop 3 time :: Double) 60
+
+-- | Converts a weekday into a double
+-- | Monday to Friday becomes 0.0 to 4.0
+getDayVal :: String -> Double
+getDayVal day = case day of
+                    "MO" -> 0.0
+                    "TU" -> 1.0
+                    "WE" -> 2.0
+                    "TH" -> 3.0
+                    "FR" -> 4.0
+
+-- | Takes a day and start/end times then generates a set of 30-minute timeslots
+getTimeSlots :: String -> String -> String -> [Time]
+getTimeSlots day start end = do
+    let dayDbl = getDayVal day
+        startDbl = getHourVal start
+        endDbl = getHourVal end
+    [Time [dayDbl, timeDbl] | timeDbl <- [startDbl, (startDbl + 0.5) .. (endDbl - 0.5)]]
+
+getNestedTimeSlots :: [Maybe T.Text] -> [Maybe T.Text] -> [Maybe T.Text] -> [[Time]]
+getNestedTimeSlots [Nothing] [Nothing] [Nothing] = [[]]
+getNestedTimeSlots days starts ends = zipWith3 (\(Just day)
+                                                 (Just start)
+                                                 (Just end) -> getTimeSlots (T.unpack day)
+                                                                            (T.unpack start)
+                                                                            (T.unpack end))
+                                               days
+                                               starts
+                                               ends
+
+ -- | URL to UofT courses (stored as JSON string)
+jsonURL :: String
+jsonURL = "https://timetable.iit.artsci.utoronto.ca/api/courses?org=&code=PRT100Y"
+
+ -- | Decode JSON string to JSON object
+getJSON :: IO (Maybe Object)
+getJSON = do
+    resp <- simpleHttp jsonURL
+    return $ decode resp
+
+-- | Extract course metadata (values) from JSON object
+-- | getMetadata :: IO (Maybe (M.HashMap k v)) -> IO [v]
+getMetadata = fmap (\(Just v) -> M.elems v)
```

Comment:
Also, pattern-matching on just a "Just" is unsafe. I'm guessing you want to return an empty list in the case of `Nothing`.